components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -2391,6 +2393,29 @@ private List<IdentityProviderProperty> filterIdentityProperties(IdentityProvider
                 }
             });
         }
+
+        identityProviderProperties.forEach(identityProviderProperty -> {
+            switch (identityProviderProperty.getName()) {
+                case IdPManagementConstants.FEDERATED_ASSOCIATION_ENABLED:
+                    federatedAssociationConfig
+                            .setEnabled(Boolean.parseBoolean(identityProviderProperty.getValue()));
+                    break;
+                case IdPManagementConstants.LOOKUP_ATTRIBUTES:
+                    String[] attributes = identityProviderProperty.getValue().split(",");
+                    federatedAssociationConfig.setLookupAttributes(attributes);
+                    break;
+            }
+        });
+
+        // For newly created idps default values will be added. However, these will be null for previously
+        // created idps and hence we need to set the default values.
+        if (federatedAssociationConfig.getLookupAttributes() == null) {
+            federatedAssociationConfig.setLookupAttributes(
+                    new String[]{IdPManagementConstants.DEFAULT_LOOKUP_ATTRIBUTE});
```

Comment:
do we need this actually? for old IDPs this config will be disabled anyway.
Also when you enable this config, one of the lookup attributes should be made mandatory, or else the update request should be failed

Reply:
I added this to provide a consistent behaviour for old and new idps. So we can enable later on in old idps as well

Reply:
Removed with [3c20442](https://github.com/wso2/carbon-identity-framework/pull/5017/commits/3c20442145ac5549fb011ce7fae823d2970cfbe9)